test/cross-browser/report.js

```diff
@@ -0,0 +1,262 @@
+        .then(function (button) {
+            return button.click();
+        });
+};
+
+var getAllHandles = function () {
+    return driver.getAllWindowHandles();
+};
+
+var getParentHandle = function () {
+    return getAllHandles()
+        .then(function (handles) {
+            return handles[0];
+        });
+};
+
+var getChildHandle = function () {
+    return getAllHandles()
+        .then(function (handles) {
+            return handles[1];
+        });
+};
+
+var getParentWindow = function () {
+    return getParentHandle()
+        .then(function (handle) {
+            return driver.switchTo().window(handle);
+        });
+};
+
+var getChildWindow = function () {
+    return getChildHandle()
+        .then(function (handle) {
+            return driver.switchTo().window(handle);
+        });
+};
+
+var loadClient = function() {
+    return driver.get('http://0.0.0.0:9999/test/cross-browser/reference.html');
+};
+
+var loadAndStartClient = function() {
+    return driver.get('http://0.0.0.0:9999/test/cross-browser/reference.html?immediate');
+};
+
+var directBrowserAway = function() {
+    return driver.get('http://0.0.0.0:9999/');
+};
+
+var createRcSocket = function() {
+    return driver.executeScript('return window.createRcSocket();');
+};
+
+var dumpClientLogger = function() {
+    return driver.sleep(200).then(function(){
+        return getChildWindow()
+            .then(function() {
+                return driver.executeScript('return window.clientLogger.buffer;');
+            })
+            .then(function(z) {
+                console.log(z);
+            })
+            .then(getParentWindow);
+    });
+};
+
+
+// TODO: pull these functions out as exports from api.js
+var socketProcess;
+var startServerSocket = function() {
+    var filePath = path.join(__dirname, '../api/web-socket.js');
+    console.log(filePath);
+
+    socketProcess = spawn('node', [filePath]);
+
+    return socketProcess;
+};
+
+var stopServerSocket = function() {
+    return socketProcess.kill('SIGINT');
+};
+
+var waitAndLog = function(z){
+    return driver.sleep(100).then(function() {
+
+        //return driver.manage().logs().get(logging.Type.BROWSER)
+        //    .then(function(entries) {
+        //        console.log(entries);
+        //        //entries.forEach(function(entry) {
+        //        //    console.log('[%s] %s', entry.level.name, entry.message);
+        //        //});
+        //    });
+
+        //(new webdriver.WebDriver.Logs(driver))
+        //    .get('browser')
+        //    .then(function (v) {
+        //        v && v.length && console.log(v);
+        //    });
+
+        console.log(z);
+
+    });
+
+    // when you want logs
+};
+
+var waitASecond = function(){
+    return driver.sleep(1000);
+};
+
+/* -----------------------------------------------------------------------------
+ * test
+ * ---------------------------------------------------------------------------*/
+
+function sharedTestcases() {
```

Comment:
Ha. I thought this was super funky when I first read it, but it works. Not sure there is any better way to do it.